power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -685,7 +707,14 @@ class Deserializer {
             elements_per_scenario < 0 ? std::reduce(counter.cbegin(), counter.cend()) : // aggregation
                 elements_per_scenario * batch_size;                                     // multiply
         handler.add_component_info(component_key_, elements_per_scenario, total_elements);
-        msg_data_offsets_.push_back(component_byte_meta);
+        // check if all scenarios does not have any map
+        bool const has_attribute_indications = std::none_of(component_byte_meta.cbegin(), component_byte_meta.cend(),
+                                                            [](auto const& x) { return x.has_map; });
```

Comment:
maybe more elaborate for the reader
```suggestion
        bool const only_values_in_data = std::none_of(component_byte_meta.cbegin(), component_byte_meta.cend(),
                                                      [](auto const& x) { return x.has_map; });
        bool const only_attribute_indications = !only_values_in_data;
```

Reply:
I do not get your intention here.

Reply:
The logic also seems to be inverted.

The name is as we said indicated in the https://github.com/PowerGridModel/power-grid-model/pull/895#discussion_r1991520332

Reply:
My intention was that I, as a reader of the code, do not see the direct relationship between `none_of(x, has_map)` and `has_attribute_indications`. You're right in that I made a typo, but I still had to read this code multiple times before I understood what the intention is.

Reply:
The first line you only change the name of the variable. What is the intension for your second line?

Reply:
resolved cfr. discussion